README.tfdoc.hcl

```
@@ -0,0 +1,1265 @@
+            description = <<-END
+              Domain prefix or fully-qualified domain name of the Cognito user pool.
+            END
+          }
+
+          attribute "user_pool_client_id" {
+            required    = true
+            type        = string
+            description = <<-END
+              Domain prefix or fully-qualified domain name of the Cognito user
+              pool.
+            END
+          }
+
+          attribute "authentication_request_extra_params" {
+            type        = map(string)
+            description = <<-END
+              Query parameters to include in the redirect request to the
+              authorization endpoint. Accepts a maximum of 10 extra parameters.
+            END
+          }
+
+          attribute "on_unauthenticated_request" {
+            type        = string
+            description = <<-END
+              Behavior if the user is not authenticated. Valid values are `deny`,
+              `allow` and `authenticate`.
+            END
+          }
+
+          attribute "scope" {
+            type        = string
+            description = <<-END
+              Set of user claims to be requested from the IdP.
+            END
+          }
+
+          attribute "session_cookie_name" {
+            type        = string
+            description = <<-END
+              Name of the cookie used to maintain session information.
+            END
+          }
+
+          attribute "session_timeout" {
+            type        = number
+            description = <<-END
+              Maximum duration of the authentication session, in seconds.
+            END
+          }
+        }
+
+        attribute "authenticate_oidc" {
+          type        = object(authenticate_oidc)
+          default     = []
+          description = <<-END
+            Configuration block for an identity provider that is compliant with
+            OpenID Connect (OIDC). Specify only when `type` is
+            `authenticate-oidc`.
+          END
+
+          attribute "authorization_endpoint" {
+            required    = true
+            type        = string
+            description = <<-END
+              Authorization endpoint of the IdP.
+            END
+          }
+
+          attribute "client_id" {
+            required    = true
+            type        = string
+            description = <<-END
+              OAuth 2.0 client identifier.
+            END
+          }
+
+          attribute "client_secret" {
+            required    = true
+            type        = string
+            description = <<-END
+              OAuth 2.0 client secret.
+            END
+          }
+
+          attribute "issuer" {
+            required    = true
+            type        = string
+            description = <<-END
+              OIDC issuer identifier of the IdP.
+            END
+          }
+
+          attribute "authentication_request_extra_params" {
+            type        = map(string)
```

```suggestion
            type        = list(authentication_request_extra_param)
```

This is a list (or a set) of objects, please add the attributes also. For details please see https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/lb_listener_rule